crates/sargon/src/system/sargon_os/sargon_os_transactions.rs

```diff
@@ -0,0 +1,382 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Submit Transaction
+// ==================
+#[uniffi::export]
+impl SargonOS {
+    /// Submits a notarized transaction payload to the network.
+    pub async fn submit_transaction(
+        &self,
+        notarized_transaction: NotarizedTransaction,
+    ) -> Result<IntentHash> {
```

Comment:
Might remove this exposed function, and just leave `submit_compiled_transaction` instead

Reply:
I would say the other way around, remove the `submit_compiled_transaction` and leave the compiling of the NotarizedTransaction to sargon, which already happens [here](https://github.com/radixdlt/sargon/blob/23d99bd21cffb6170d2caaa1aec5eea994c1e6c4/crates/sargon/src/gateway_api/models/logic/request/transaction/submit/transaction_submit.rs#L5).